pkg/test/framework/featurelabels/features.go

```diff
@@ -0,0 +1,18 @@
+package featurelabels
+
+type Security struct {
+	MTLS MTLS
+}
+
+type MTLS struct {
+}
+
+type Observability struct {
```

Comment:
Not sure about this approach. If we use random structs to represent "features", then basically anything could be a feature. Wouldn't we lose all type checking, since we'd have to pass them around as `struct{}` everywhere?

I was thinking that we would just have them all as constant strings in some file. We would have to make sure that we spell everything correctly, but we could organize them so they're in the same file, making it easier to catch type-os.

Reply:
I'd prefer to keep them hierarchical for usability, but I see what you mean about type safety.  Let me see if I can use go's pseudo inheritance to solve this...

Reply:
It's not just about usability, it's also about complexity. What exactly does representing the feature hierarchy this way afford you? The way this is structured, it will be exhaustive to read and maintain compared to raw strings, so there should be some pretty strong justification for the added complexity.